crates/uv/tests/pip_sync.rs

```diff
@@ -1098,10 +1102,8 @@ fn mismatched_name() -> Result<()> {
     ----- stdout -----
 
     ----- stderr -----
-    Resolved 1 package in [TIME]
-    Downloaded 1 package in [TIME]
-    error: Failed to install: foo-2.0.1-py3-none-any.whl (foo==2.0.1 (from file://[TEMP_DIR]/foo-2.0.1-py3-none-any.whl))
-      Caused by: Wheel package name does not match filename: tomli != foo
+    error: Failed to read: foo @ file://[TEMP_DIR]/foo-2.0.1-py3-none-any.whl
+      Caused by: The .dist-info directory tomli-2.0.1 does not start with the normalized package name: tomli-2.0.1
```

Comment:
Slightly worse, maybe (not sure why `tomli-2.0.1` is repeated here, I will look into it, might be an existing bug).

I think this is because we now validate the metadata to get `Requires-Python` enforcement, whereas before we ignored the metadata and only validated the wheel when we installed.